src/services/ivaDatabaseService.ts

```diff
@@ -0,0 +1,66 @@
+import DynamoDB, { DocumentClient } from "aws-sdk/clients/dynamodb";
+import { IDBConfig } from "../models";
+import { Configuration } from "../utils/Configuration";
+import { PromiseResult } from "aws-sdk/lib/request";
+import {
+  EUVehicleCategory,
+  InspectionType,
+  VehicleType,
+} from "@dvsa/cvs-type-definitions/types/iva/defects/get";
+
+export class IvaDatabaseService {
+  private readonly tableName: string;
+  private static dbClient: DocumentClient;
+
+  constructor() {
+    const config: IDBConfig = Configuration.getInstance().getDynamoDBConfig();
+    this.tableName = config.ivaDefects.table;
+    if (!IvaDatabaseService.dbClient) {
+      IvaDatabaseService.dbClient = new DynamoDB.DocumentClient(
+        config.ivaDefects,
+      );
+    }
+  }
+
+  public async getDefectsByManualId(
+    manualId: string,
+  ): Promise<Array<Record<string, any>>> {
+    return await this.queryAllData({
+      TableName: this.tableName,
+      FilterExpression: "euVehicleCategories_0 = :manualId",
+      ExpressionAttributeValues: {
+        ":manualId": manualId,
+      },
+    });
+  }
+
+  public getDefectsByCriteria(
+    vehicleType: VehicleType | null,
```

Comment:
It would otherwise default to the first value in the VehicleType type and you'd never be able to tell whether or not the client had actually supplied a value in the first place.